src/pebble.py

```diff
@@ -219,6 +219,28 @@ def _get_synapse_config(container: ops.model.Container) -> dict:
         raise PebbleServiceError(str(exc)) from exc
 
 
+def _get_mas_config(container: ops.model.Container) -> dict:
+    """Get the current MAS configuration.
+
+    Args:
+        container: Synapse container.
+
+    Returns:
+        dict: MAS configuration.
+
+    Raises:
+        PebbleServiceError: if something goes wrong while interacting with Pebble.
+    """
+    try:
+        config = container.pull(MAS_CONFIGURATION_PATH).read()
+        return yaml.safe_load(config)
+    except ops.pebble.PathError as exc:
+        # If the MAS config has not been created, we return an empty dict to trigger a replan
```

Comment:
When could this happen?

Reply:
The 2 most likely cases are: 
When the charm first starts, or when the config is manually deleted for whatever reason